tck/src/main/java/org/eclipse/microprofile/lra/tck/service/spi/LRARecoveryService.java

```diff
@@ -0,0 +1,88 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * You may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *******************************************************************************/
+package org.eclipse.microprofile.lra.tck.service.spi;
+
+import java.net.URI;
+
+/**
+ * This SPI is providing the implementation with the ability to define what actions need to be taken
+ * in case the TCK needs to wait for the delivery some eventually consistent LRA actions (e.g. Compensate or
+ * Complete calls). See the individual methods for details.
+ */
+public interface LRARecoveryService {
+
+    /**
+     * <p>
+     * This method is utilized by the TCK to wait for the delivery of all participant callbacks
+     * (Complete, Compensate, AfterLRA, ...) for a given LRA. When the LRA is asked to perform some enclosing
+     * eventually consistent action (close/cancel) which results into subsequent calls to every Complete/Compensate
+     * method of individual enlisted participants and this method is invoked it shouldn't return before
+     * all of the responses from the participant callbacks are successfully received. This allows the implementation
+     * to define what implementation specific actions can be taken so the TCK can avoid waiting for the eventual 
+     * callbacks and responses. In the basic case, simply waiting in this method would be equivalent to the behavior
+     * of the TCK without the call to this method. When this method finishes the participant methods invocation results
+     * are verified so the implementation must make sure that this method doesn't return before the validation
+     * can happen.
+     * </p>
+     * 
+     * Example:
+     * <ol>
+     *   <li>call to close LRA with id lraId (Complete calls are starting to be performed)</li>
+     *   <li>call to {@link LRARecoveryService#waitForCallbacks(URI)} with lraId -- wait for all Complete calls to be
+     * delivered and optionally perform some implementation specific actions to make the calls return faster</li>
+     *   <li>call to {@link LRARecoveryService#waitForCallbacks(URI)} returns</li>
+     *   <li>the TCK test verifies that all Complete methods have been called</li>
+     * </ol>
+     *
+     * @param lraId the LRA id of the LRA which callbacks should be delivered when this method exits
+     * @throws Exception in case of any error. The exception will be propagated to fail the enclosing TCK test.
+     */
+    void waitForCallbacks(URI lraId) throws Exception;
```

Comment:
Can we use a specific exception to indicate that callbacks were not successful and in the tests catch it and assert fail. Using Exception will produce stack traces in the tests and make it harder to differentiate other TCK exceptions from callback failures.